src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -754,4 +754,20 @@ object messages {
            |${"square(ints: _*)          // res1: List[Int] = List(4, 9, 16)"}
            |""".stripMargin
   }
+
+  case class IllegalLiteral()(implicit ctx: Context) extends Message(28) {
+    val kind = "Syntax"
+    val msg = "illegal literal"
+    val explanation =
+      hl"""|Expected a valid literal.
```

Comment:
I would skip this line, the error already says "illegal literal" right? If you think it should be kept, you could bake it into the sentence where you present the legal literals in Scala.